python/lib/sift_py/data_import/ch10.py

```diff
@@ -0,0 +1,111 @@
+import json
+from typing import Any, Dict, Optional
+
+import requests
+
+from sift_py.data_import.config import CsvConfig
+from sift_py.data_import.csv import CsvUploadService
+from sift_py.data_import.status import DataImportService
+from sift_py.data_import.time_format import TimeFormatType
+
+
+class BaseCh10File:
+    """
+    Base class for uploading IRIG Chapter 10/Chapter 11 files.
+
+    Implement a concrete version of this class that parses a ch10 stream and returns
+    a csv row of data on each iteration.
+
+    Note: The first iteration must populate the csv_config_data_columns attribute
+    that is the data_columns entry in the CsvConfig.
+    """
+
+    csv_config_data_columns: Dict[int, dict]
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        raise NotImplementedError
+
+
+class Ch10UploadService(CsvUploadService):
+    """Service to upload ch10 files."""
+
+    def upload_ch10(
+        self,
+        ch10_file: BaseCh10File,
+        asset_name: str,
+        time_format: TimeFormatType = TimeFormatType.ABSOLUTE_UNIX_NANOSECONDS,
+        run_name: Optional[str] = None,
+        run_id: Optional[str] = None,
+    ) -> DataImportService:
+        """
+        Uploads the ch10 file to the specified asset.
+
+        Override `time_format` to specify the time data format. Default is `TimeFormatType.ABSOLUTE_UNIX_NANOSECONDS`.
+        Override `run_name` to specify the name of the run to create for this data. Default is None.
+        Override `run_id` to specify the id of the run to add this data to. Default is None.
+        """
+        # Trigger the first packet read to get the row headers
+        # and csv_config_data_columns.
+        next(ch10_file)
+
+        config_info: Dict[str, Any] = {
+            "asset_name": asset_name,
+            "first_data_row": 1,
+            "time_column": {
+                "format": time_format,
+                "column_number": 1,
+            },
+            "data_columns": ch10_file.csv_config_data_columns,
```

Comment:
Is it possible for users to get to this point and have not provided `csv_config_data_columns`? (And can we raise an exception if it is possible? Otherwise no worries if it's not.)